### PR TITLE
增加uart print到master

### PR DESCRIPTION
**Pull request by ts05593818**
uart_print("%d",i);
未測試

### CODE REVIEW

**Comment by ts05593818** [comment]
新增修改過的uart_print.c uart_print.h main.h

File: source/uart_print.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            uart_put_c(data[count++]);
+            continue;
+        }
+
+        data++;         // backward a char
+
+            switch (data[count]) {
+            case 'd' : {
+                int dd;
+                dd=va_arg(arg_ptr,int);
```
Comment (a910184) [inline source/uart_print.c]: image
測試code
result
image
'255' should be character not value

Comment (a910184) [reply source/uart_print.c]: @ts05593818

File: source/uart_print.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                }
+                break;
+            };
+            case 'x' : {
+                int xx;
+                char hex_char[17]="0123456789ABCDEF";    // HEX string
+                xx=va_arg(arg_ptr,int);
+                uart_put_c(hex_char[high_bits(xx)]);
+                uart_put_c(hex_char[low_bits(xx)]);
+                break;
```
Comment (a910184) [inline source/uart_print.c:66]: only support value less than 255??

Comment (a910184) [reply source/uart_print.c:66]: @ts05593818

**Pull request merged**